File: library/src/main/java/com/example/library/anim/SlideFormLeftAnim.java
package com.example.library.anim;

import android.animation.Animator;
import android.animation.ObjectAnimator;
import android.support.v4.view.animation.LinearOutSlowInInterpolator;
import android.view.View;

import com.example.library.config.Constant;

/**
 * 从左边滑进来的动画
 * Created by 陈健宇 at 2019/7/10
 */
public class SlideFormLeftAnim implements IAnim {

    @Override
    public void applyAnimation(View itemView) {
        Animator translationFromLeft = ObjectAnimator.ofFloat(itemView, "translationX", -itemView.getRootView().getWidth(), 0);
        translationFromLeft.setInterpolator(new LinearOutSlowInInterpolator());
        translationFromLeft.setDuration(Constant.ITEM_ANIM_TIME);
        translationFromLeft.start();
    }

}

File: library/src/main/java/com/example/library/anim/SlideFormLeftAnim.kt
package com.example.library.anim

import android.animation.Animator
import android.animation.ObjectAnimator
import android.view.View

import com.example.library.config.Constant

/**
 * 从左边滑进来的动画
 * Created by 陈健宇 at 2019/7/10
 */
class SlideFormLeftAnim : IAnim {

    override fun applyAnimation(itemView: View) {
        val translationFromLeft = ObjectAnimator.ofFloat(itemView, "translationX", -itemView.rootView.width.toFloat(), 0f)
        translationFromLeft.run{
            duration = Constant.ITEM_ANIM_TIME.toLong()
            start()
        }
    }

}
